examples/tree_1d_dgsem/elixir_shallowwater_multilayer_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,199 @@
+                FluxHydrostaticReconstruction(flux_nonconservative_ersing_etal,
+                                              hydrostatic_reconstruction_ersing_etal))
+basis = LobattoLegendreBasis(3)
+
+indicator_sc = IndicatorHennemannGassnerShallowWater(equations, basis,
+                                                     alpha_max = 0.5,
+                                                     alpha_min = 0.001,
+                                                     alpha_smooth = true,
+                                                     variable = waterheight_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(indicator_sc;
+                                                 volume_flux_dg = volume_flux,
+                                                 volume_flux_fv = surface_flux)
+solver = DGSEM(basis, surface_flux, volume_integral)
+###############################################################################
+# Get the TreeMesh and setup a periodic mesh
+
+# The domain of interest [0.0, 100.0] is extended towards -28.0 to match the positions of the 
+# disctontinuities with TreeMesh.
+coordinates_min = -28.0
+coordinates_max = 100.0
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 7,
+                n_cells_max = 10_000,
+                periodicity = true)
+
+# create the semi discretization object
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 10.0)
+ode = semidiscretize(semi, tspan)
+
+#################################################################################
+# Workaround to set a discontinuous water and bottom topography for
+# debugging and testing. Essentially, this is a slight augmentation of the
+# `compute_coefficients` where the `x` node value passed here is slightly
+# perturbed to the left / right in order to set a true discontinuity that avoids
+# the doubled value of the LGL nodes at a particular element interface.
+#
+# Note! The errors from the analysis callback are not important but the error
+# for this lake at rest test case `∑|H0-(h+b)|` should be near machine roundoff.
+
+# point to the data we want to augment
+u = Trixi.wrap_array(ode.u0, semi)
+# reset the initial condition
+for element in eachelement(semi.solver, semi.cache)
+    for i in eachnode(semi.solver)
+        x_node = Trixi.get_node_coords(semi.cache.elements.node_coordinates, equations,
+                                       semi.solver, i, element)
+        # We know that the discontinuity is a vertical line. Slightly augment the x value by a factor
+        # of unit roundoff to avoid the repeted value from the LGL nodes at at interface.
+        if i == 1
+            x_node = SVector(nextfloat(x_node[1]))
+        elseif i == nnodes(semi.solver)
+            x_node = SVector(prevfloat(x_node[1]))
+        end
```

Comment:
This workaround should no longer be necessary because this `prevfloat` / `nextfloat` strategy is implemented in Trixi for the 1D mesh, see [this function](https://github.com/trixi-framework/Trixi.jl/blob/b1a84a63298966b67491dacac78e785e0dbfb36c/src/solvers/dg.jl#L700).

Reply:
This has been removed in 3d6d89a9381cd544f4ee470550c38ee47df42bf1.
I did not found any other occurences in `TrixiShallowWater` or `Trixi`